central/networkbaseline/manager/manager_impl.go

```diff
@@ -313,15 +313,36 @@ func (m *manager) ProcessDeploymentCreate(deploymentID, _, clusterID, _ string)
 	return m.processDeploymentCreate(deploymentID, clusterID)
 }
 
+func (m *manager) deleteDeploymentFromBaselines(deploymentID string) error {
+	modifiedDeployments := set.NewStringSet()
+	for id, baseline := range m.baselinesByDeploymentID {
+		if ok, peer := baseline.GetPeer(deploymentID); ok {
+			delete(baseline.BaselinePeers, peer)
+			modifiedDeployments.Add(id)
+		}
+
+		if ok, forbiddenPeer := baseline.GetForbiddenPeer(deploymentID); ok {
+			delete(baseline.ForbiddenPeers, forbiddenPeer)
+			modifiedDeployments.Add(id)
+		}
+	}
+
+	if err := m.persistNetworkBaselines(modifiedDeployments, nil); err != nil {
+		return errors.Wrapf(err, "deleting baseline of deployment ID %s", deploymentID)
+	}
+
+	return nil
+}
+
 func (m *manager) processDeploymentDelete(deploymentID string) error {
 	deletingBaseline, found := m.baselinesByDeploymentID[deploymentID]
 	if !found {
 		// Most likely a repeated call to delete. Just return
 		return nil
 	}
 
-	// If the deployment is being tracked, but the baseline is not yet created, we cannot look at hte peers.  If we
-	// have a peer at all, then the peer will have been created
+	// If baseline for deleting deploynment exists, than we should look at all entries in the baseline
```

Comment:
```suggestion
	// If baseline for deleting deploynment exists, then we should look at all entries in the baseline
```
Nit: I find the comment a bit hard to understand. Making sure I understand it right:
If a baseline for this deployment exists, we remove the deployment from all peer baselines as well, correct?

So, with this PR we are fixing the inconsistencies in the Baseline, but not changing the current logic, right?
Do we have a follow up ticket to track investigation whether we actually want to change the logic?